### ISSUE TITLE
Extension host terminated unexpectedly.

### ISSUE THREAD

**Issue by ishuuwag** [mike-lischke/vscode-antlr4#44]
<img width="257" alt="screen shot 2018-08-27 at 20 39 09" src="https://user-images.githubusercontent.com/25029568/44748055-0fa30080-ab0f-11e8-8700-99667be9d486.png">
<img width="1540" alt="screen shot 2018-08-28 at 22 10 43" src="https://user-images.githubusercontent.com/25029568/44748192-732d2e00-ab0f-11e8-8f37-eedffa0e9977.png">
Reproducing steps:
Open an Antlr project with Visual Studio Code
Node version: v8.11.4
Npm version : 5.6.0
macOS 10.13.6

**Comment by sirgru** [comment]
I am getting the same error when starting the Task Runner when the extension hasn't fully loaded or when there is an error in configuration of the task runner. Worse, the error keeps repeating on my end until VS Code is not re-opened even after the error is fixed.

**Comment by mike-lischke** [comment]
Try deleting the .antlr folder in your project, which contains generated data. The error location in the sshot indicates that loading the exported ATN might be a problem. If that doesn't help please add the grammar here that causes the trouble. I can then see there's something with the ATN generation.

**Comment by ishuuwag** [comment]
It works after deleting the .antlr folder. But gives the same error if I use the same grammar. So it's probably a problem in my grammar. 
[velo.g4.txt](https://github.com/mike-lischke/vscode-antlr4/files/2367564/velo.g4.txt)

**Comment by ishuuwag** [comment]
grammar velo;

/*
 * Parser rules
 */
velo: imagedesc;
imagedesc: (baseimage)| (baseimage NEWLINE descname)
	| (baseimage NEWLINE descname NEWLINE author)
	| (baseimage NEWLINE descname NEWLINE author NEWLINE label)
	| (baseimage NEWLINE descname NEWLINE author NEWLINE label NEWLINE copy);
baseimage: BASEIMAGE NAME;
descname: DESC NAME;
author: AUTHOR NAME;
label: LABEL NAME;
copy: COPY NAME NAME;
/*
 * Lexer rules
 */

fragment A: ('A');
fragment B: ('B');
fragment C: ('C');
fragment D: ('D');
fragment E: ('E');
fragment F: ('F');
fragment G: ('G');
fragment H: ('H');
fragment I: ('I');
fragment J: ('J');
fragment K: ('K');
fragment L: ('L');
fragment M: ('M');
fragment N: ('N');
fragment O: ('O');
fragment P: ('P');
fragment Q: ('Q');
fragment R: ('R');
fragment S: ('S');
fragment T: ('T');
fragment U: ('U');
fragment V: ('V');
fragment W: ('W');
fragment X: ('X');
fragment Y: ('Y');
fragment Z: ('Z');
fragment LOWERCASE: [a-z];
fragment UPPERCASE: [A-Z];
fragment DIGIT: [0-9];

BASEIMAGE: F R O M;
DESC: N A M E;
AUTHOR: A U T H O R;
LABEL: L A B E L;
COPY: C O P Y;
RUN: R U N;
WORKDIR: W O R K D I R;
NAME: (LOWERCASE | UPPERCASE | DIGIT |)+;
NEWLINE: ('\r'? '\n' | '\r')+;

WS: [ \t\r\n]+ -> skip; // skip spaces, tabs, newlines

**Comment by mike-lischke** [comment]
I found the issue in your grammar:

    NAME: (LOWERCASE | UPPERCASE | DIGIT |)+;

You didn't see the warning from ANTLR because vscode crashed, but that's what I got:

<img width="729" alt="bildschirmfoto 2018-10-14 um 13 36 48" src="https://user-images.githubusercontent.com/2398704/46916106-52814080-cfb6-11e8-86ab-9618f8ad0b5f.png">

This causes trouble in the underlying TS runtime, when optimizing the generated ATN. The rule itself is however wrong and can lead to endless recursions, trying to match nothing one or more times. Remove the empty alt and you are fine. If `NAME` can be empty then make it optional in the caller, e.g.:

    baseimage: BASEIMAGE NAME?;
